apps/studio/src/server/modules/page/__tests__/page.router.test.ts

```diff
@@ -0,0 +1,1206 @@
+        pageId: expect.any(String),
+      })
+      expect(actual).toMatchObject({
+        ...expectedPageArgs,
+        content: createDefaultPage({ layout: "content" }),
+      })
+    })
+
+    it("should create a page in folder successfully", async () => {
+      // Arrange
+      const { site, folder } = await setupFolder()
+      const expectedPageArgs = {
+        siteId: site.id,
+        title: "Test Page",
+        permalink: "test-page",
+      }
+
+      // Act
+      const result = await caller.createPage({
+        ...expectedPageArgs,
+        layout: "content",
+        folderId: Number(folder.id),
+      })
+
+      // Assert
+      const actual = await db
+        .selectFrom("Resource")
+        .innerJoin("Blob", "Resource.draftBlobId", "Blob.id")
+        .where("Resource.id", "=", result.pageId)
+        .select([
+          "title",
+          "permalink",
+          "type",
+          "siteId",
+          "Blob.content",
+          "parentId",
+        ])
+        .executeTakeFirstOrThrow()
+      expect(result).toMatchObject({
+        pageId: expect.any(String),
+      })
+      expect(actual).toMatchObject({
+        ...expectedPageArgs,
+        parentId: folder.id,
+        content: createDefaultPage({ layout: "content" }),
+      })
+    })
+
+    it("should throw 400 if folderId does not exist", async () => {
+      // Arrange
+      const { site } = await setupSite()
+      const expectedPageArgs = {
+        siteId: site.id,
+        title: "Test Page",
+        permalink: "test-page",
+        folderId: 999999, // should not exist
+      }
+
+      // Act
+      const result = caller.createPage({
+        ...expectedPageArgs,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "BAD_REQUEST",
+          message: "Folder not found or folderId is not a folder",
+        }),
+      )
+    })
+
+    it("should throw 400 if folderId is not a Folder resource", async () => {
+      // Arrange
+      const { site, page } = await setupPageResource({ resourceType: "Page" })
+      const expectedPageArgs = {
+        siteId: site.id,
+        title: "Test Page",
+        permalink: "test-page",
+        folderId: Number(page.id),
+      }
+
+      // Act
+      const result = caller.createPage({
+        ...expectedPageArgs,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "BAD_REQUEST",
+          message: "Folder not found or folderId is not a folder",
+        }),
+      )
+    })
+
+    // TODO: Implement tests when permissions are implemented
+    it.skip("should throw 403 if user does not have access to site", async () => {})
+
+    it.skip("should throw 403 if user does not have write access to folder", async () => {})
+
+    it.skip("should throw 403 if user does not have write access to root", async () => {})
+  })
+
+  describe("getRootPage", () => {
```

Comment:
low value case but should we check behaviour of what happens when the `parentId` of rootpage is non-null?

technically this should return a 5xx but i think it'll pass at present. although it's worth noting that actually users don't get to edit this property so it's really really really very minor and i'm happy to skip